src/main/java/opendcs/util/sql/WrappedConnection.java

```diff
@@ -0,0 +1,394 @@
+            throws SQLException {
+        return realConnection.prepareStatement(sql, resultSetType, resultSetConcurrency);
+    }
+
+    @Override
+    public PreparedStatement prepareStatement(String sql, int resultSetType, int resultSetConcurrency,
+            int resultSetHoldability) throws SQLException {
+        return realConnection.prepareStatement(sql, resultSetType, resultSetConcurrency, resultSetHoldability);
+    }
+
+    @Override
+    public void releaseSavepoint(Savepoint savepoint) throws SQLException {
+        realConnection.releaseSavepoint(savepoint);
+    }
+
+    @Override
+    public void rollback() throws SQLException {
+        realConnection.rollback();
+    }
+
+    @Override
+    public void rollback(Savepoint savepoint) throws SQLException {
+        realConnection.rollback(savepoint);
+    }
+
+    @Override
+    public void setAutoCommit(boolean autoCommit) throws SQLException {
+        realConnection.setAutoCommit(autoCommit);
+    }
+
+    @Override
+    public void setCatalog(String catalog) throws SQLException {
+        realConnection.setCatalog(catalog);
+    }
+
+    @Override
+    public void setClientInfo(Properties properties) throws SQLClientInfoException {
+        realConnection.setClientInfo(properties);
+    }
+
+    @Override
+    public void setClientInfo(String name, String value) throws SQLClientInfoException {
+        realConnection.setClientInfo(name, value);
+    }
+
+    @Override
+    public void setHoldability(int holdability) throws SQLException {
+        realConnection.setHoldability(holdability);
+    }
+
+    @Override
+    public void setNetworkTimeout(Executor executor, int milliseconds) throws SQLException {
+        realConnection.setNetworkTimeout(executor, milliseconds);
+    }
+
+    @Override
+    public void setReadOnly(boolean readOnly) throws SQLException {
+        realConnection.setReadOnly(readOnly);   
+    }
+
+    @Override
+    public Savepoint setSavepoint() throws SQLException {
+        return realConnection.setSavepoint();
+    }
+
+    @Override
+    public Savepoint setSavepoint(String name) throws SQLException {
+        return realConnection.setSavepoint(name);
+    }
+
+    @Override
+    public void setSchema(String schema) throws SQLException {
+        realConnection.setSchema(schema);
+        
+    }
+
+    @Override
+    public void setTransactionIsolation(int level) throws SQLException {
+        realConnection.setTransactionIsolation(level);
+    }
+
+    @Override
+    public void setTypeMap(Map<String, Class<?>> map) throws SQLException {
+        realConnection.setTypeMap(map);
+    }
+
+    public CompositeData asCompositeData() throws OpenDataException
+    {
+        String[] itemNames = new String[1];
+        Object[] values = new Object[1];
+        itemNames[0] = "Lifetime";
+        values[0] = Duration.between(start, end == null ? ZonedDateTime.now() : end)
```

Comment:
seems like it would be good to log here if end is null and the duration between start and now has gone past some threshold.

Reply:
Yeah, but the bean info doesn't currently even show up. I'll fix up the JMX values in a later PR.